server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalSum.java

```diff
@@ -70,6 +70,13 @@ public double getValue() {
         return sum;
     }
 
+    @Override
+    public boolean hasValue() {
+        // TODO this could be incorrect... e.g. +1 + -1
+        // Think we'll have to serialize count if we want to fix this...
```

Comment:
Another idea which might impact backwards compatibility would be to use an out of range value (Infinity or NaN) and check that instead of the count.

Reply:
Just took a look and that turns out to be tricky (which is a shame, after you mentioned it I agreed that would be best).  Sum agg and a few others use Kahan summation for improved numerical stability... but part of that explicitly handles when the sum goes non-finite as part of the compensation scheme.  So I think we're stuck with recording counts like the Avg does.